### PR TITLE
Issue 2197: Allow to bypass starting requirements

### PR DESCRIPTION
**Pull request by nathan-mittelette**
### Description

In this PR, I added a new environment variable called **system.requirement.check.enabled** that allow users to disable Dependency-Track requirements startup check.

### Addressed Issue

Fixes DependencyTrack/dependency-track#2197

### Checklist

- [ ] I have read and understand the [contributing guidelines](https://github.com/DependencyTrack/dependency-track/blob/github-templates/CONTRIBUTING.md#pull-requests)
- [ ] This PR introduces new or alters existing behavior, and I have updated the [documentation](https://github.com/DependencyTrack/dependency-track/tree/master/docs/_docs) accordingly

### CODE REVIEW

**Comment by valentijnscholten** [comment]
I like this as it allows users to run DT as they see fit. In my experience 2G also works fine for smaller instances which allows DT to run on cheaper VM instances.

**Comment by valentijnscholten** [comment]
The word "bypass" feels weird in the context of a security product. How about DISABLE_SYSTEM_REQUIREMENT_CHECK?

**Comment by nscuro** [comment]
@stevespringett Any hard objections against this? Added my 2ct in https://github.com/DependencyTrack/dependency-track/issues/2197#issuecomment-1330927879

If we decide to adapt this, there should be a reference to the config option on the system requirements page(s) in the documentation:
* https://docs.dependencytrack.org/getting-started/deploy-docker/#container-requirements-api-server
* https://docs.dependencytrack.org/getting-started/deploy-exewar/#requirements

**Comment by nathan-mittelette** [comment]
> The word "bypass" feels weird in the context of a security product. How about DISABLE_SYSTEM_REQUIREMENT_CHECK?

I agree with you, I updated the property name.

**Comment by nathan-mittelette** [comment]
@nscuro 

> If we decide to adapt this, there should be a reference to the config option on the system requirements page(s) in the documentation:
> 
> * https://docs.dependencytrack.org/getting-started/deploy-docker/#container-requirements-api-server
> * https://docs.dependencytrack.org/getting-started/deploy-exewar/#requirements

I updated the documentation. Do these changes work for you?

File: src/main/resources/application.properties [hunk 1]
```diff
@@ -317,6 +317,11 @@ alpine.oidc.team.synchronization=false
 # will most likely need to be configured.
 alpine.oidc.teams.claim=groups
 
+# Optional
+# Define whether system requirement check is disable.
+# The default value is false.
+disable.system.requirement.check=false
```
Comment (valentijnscholten) [inline src/main/resources/application.properties:323]: don't want to nitpick, but I suggest {{system.requirement.check.enabled=true}}, similar to the CORS setting etc.

**Comment by stevespringett** [comment]
@nscuro 
> Any hard objections against this? Added my 2ct in

No objections. But I would add a WARN statement on startup stating that its disabled and provide a URL to the docs where the minimum requirements can be found.

My concern is that Person A disables the check because they have a small DT instance without a lot of activity. After a year or more, Person B takes over managing DT which has now greatly increased the amount of BOM consumption and they're running into performance issues as a result. Person B doesn't have the backstory, but if we have a WARN in the log, they may see that.

**Comment by nscuro** [comment]
> After a year or more, Person B takes over managing DT which has now greatly increased the amount of BOM consumption and they're running into performance issues as a result.

To be fair this also can happen when the instance starts out with 4GB, but usage is ramped up to a point where 8 or 16GB would be more appropriate. But I agree that at the lower end of specs the exhaustion of resources is more likely to occur, and a warning is a good thing to have.

**Review by nscuro — APPROVED** [review]
Thanks @nathan-mittelette! 🚀

**Comment by mulder999** [comment]
To add to the thread, I noticed the max memory size is checked statically at startup. With current VM memory ballooning technics this cannot work as memory would increase with actual usage. Also the max memory gives no indication whether this memory could be requested or not by the application (thinking container world here). I believe it would be more suitable to allocate the memory needed along the way and not bother with this check at all, those that have memory issues will notice it the hardway anyway... At least now this check is optional.

**Pull request merged**